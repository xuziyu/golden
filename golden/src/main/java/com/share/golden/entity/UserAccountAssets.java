package com.share.golden.entity;

import java.math.BigDecimal;

public class UserAccountAssets {
    private BigDecimal accountId;

    private String userId;

    private String accountBalance;

    private String accountFreeze;

    private String accountDue;

    private String accountPaid;

    private String totalMoney;

    public BigDecimal getAccountId() {
        return accountId;
    }

    public void setAccountId(BigDecimal accountId) {
        this.accountId = accountId;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public String getAccountBalance() {
        return accountBalance;
    }

    public void setAccountBalance(String accountBalance) {
        this.accountBalance = accountBalance == null ? null : accountBalance.trim();
    }

    public String getAccountFreeze() {
        return accountFreeze;
    }

    public void setAccountFreeze(String accountFreeze) {
        this.accountFreeze = accountFreeze == null ? null : accountFreeze.trim();
    }

    public String getAccountDue() {
        return accountDue;
    }

    public void setAccountDue(String accountDue) {
        this.accountDue = accountDue == null ? null : accountDue.trim();
    }

    public String getAccountPaid() {
        return accountPaid;
    }

    public void setAccountPaid(String accountPaid) {
        this.accountPaid = accountPaid == null ? null : accountPaid.trim();
    }

    public String getTotalMoney() {
        return totalMoney;
    }

    public void setTotalMoney(String totalMoney) {
        this.totalMoney = totalMoney == null ? null : totalMoney.trim();
    }
}