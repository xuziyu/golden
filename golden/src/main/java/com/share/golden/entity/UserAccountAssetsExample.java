package com.share.golden.entity;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

public class UserAccountAssetsExample {
    protected String orderByClause;

    protected boolean distinct;

    protected List<Criteria> oredCriteria;

    public UserAccountAssetsExample() {
        oredCriteria = new ArrayList<Criteria>();
    }

    public void setOrderByClause(String orderByClause) {
        this.orderByClause = orderByClause;
    }

    public String getOrderByClause() {
        return orderByClause;
    }

    public void setDistinct(boolean distinct) {
        this.distinct = distinct;
    }

    public boolean isDistinct() {
        return distinct;
    }

    public List<Criteria> getOredCriteria() {
        return oredCriteria;
    }

    public void or(Criteria criteria) {
        oredCriteria.add(criteria);
    }

    public Criteria or() {
        Criteria criteria = createCriteriaInternal();
        oredCriteria.add(criteria);
        return criteria;
    }

    public Criteria createCriteria() {
        Criteria criteria = createCriteriaInternal();
        if (oredCriteria.size() == 0) {
            oredCriteria.add(criteria);
        }
        return criteria;
    }

    protected Criteria createCriteriaInternal() {
        Criteria criteria = new Criteria();
        return criteria;
    }

    public void clear() {
        oredCriteria.clear();
        orderByClause = null;
        distinct = false;
    }

    protected abstract static class GeneratedCriteria {
        protected List<Criterion> criteria;

        protected GeneratedCriteria() {
            super();
            criteria = new ArrayList<Criterion>();
        }

        public boolean isValid() {
            return criteria.size() > 0;
        }

        public List<Criterion> getAllCriteria() {
            return criteria;
        }

        public List<Criterion> getCriteria() {
            return criteria;
        }

        protected void addCriterion(String condition) {
            if (condition == null) {
                throw new RuntimeException("Value for condition cannot be null");
            }
            criteria.add(new Criterion(condition));
        }

        protected void addCriterion(String condition, Object value, String property) {
            if (value == null) {
                throw new RuntimeException("Value for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value));
        }

        protected void addCriterion(String condition, Object value1, Object value2, String property) {
            if (value1 == null || value2 == null) {
                throw new RuntimeException("Between values for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value1, value2));
        }

        public Criteria andAccountIdIsNull() {
            addCriterion("\"ACCOUNT_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andAccountIdIsNotNull() {
            addCriterion("\"ACCOUNT_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andAccountIdEqualTo(BigDecimal value) {
            addCriterion("\"ACCOUNT_ID\" =", value, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdNotEqualTo(BigDecimal value) {
            addCriterion("\"ACCOUNT_ID\" <>", value, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdGreaterThan(BigDecimal value) {
            addCriterion("\"ACCOUNT_ID\" >", value, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdGreaterThanOrEqualTo(BigDecimal value) {
            addCriterion("\"ACCOUNT_ID\" >=", value, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdLessThan(BigDecimal value) {
            addCriterion("\"ACCOUNT_ID\" <", value, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdLessThanOrEqualTo(BigDecimal value) {
            addCriterion("\"ACCOUNT_ID\" <=", value, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdIn(List<BigDecimal> values) {
            addCriterion("\"ACCOUNT_ID\" in", values, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdNotIn(List<BigDecimal> values) {
            addCriterion("\"ACCOUNT_ID\" not in", values, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdBetween(BigDecimal value1, BigDecimal value2) {
            addCriterion("\"ACCOUNT_ID\" between", value1, value2, "accountId");
            return (Criteria) this;
        }

        public Criteria andAccountIdNotBetween(BigDecimal value1, BigDecimal value2) {
            addCriterion("\"ACCOUNT_ID\" not between", value1, value2, "accountId");
            return (Criteria) this;
        }

        public Criteria andUserIdIsNull() {
            addCriterion("\"USER_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andUserIdIsNotNull() {
            addCriterion("\"USER_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andUserIdEqualTo(String value) {
            addCriterion("\"USER_ID\" =", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotEqualTo(String value) {
            addCriterion("\"USER_ID\" <>", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdGreaterThan(String value) {
            addCriterion("\"USER_ID\" >", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"USER_ID\" >=", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLessThan(String value) {
            addCriterion("\"USER_ID\" <", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLessThanOrEqualTo(String value) {
            addCriterion("\"USER_ID\" <=", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLike(String value) {
            addCriterion("\"USER_ID\" like", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotLike(String value) {
            addCriterion("\"USER_ID\" not like", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdIn(List<String> values) {
            addCriterion("\"USER_ID\" in", values, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotIn(List<String> values) {
            addCriterion("\"USER_ID\" not in", values, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdBetween(String value1, String value2) {
            addCriterion("\"USER_ID\" between", value1, value2, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotBetween(String value1, String value2) {
            addCriterion("\"USER_ID\" not between", value1, value2, "userId");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceIsNull() {
            addCriterion("\"ACCOUNT_BALANCE\" is null");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceIsNotNull() {
            addCriterion("\"ACCOUNT_BALANCE\" is not null");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceEqualTo(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" =", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceNotEqualTo(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" <>", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceGreaterThan(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" >", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceGreaterThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" >=", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceLessThan(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" <", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceLessThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" <=", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceLike(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" like", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceNotLike(String value) {
            addCriterion("\"ACCOUNT_BALANCE\" not like", value, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceIn(List<String> values) {
            addCriterion("\"ACCOUNT_BALANCE\" in", values, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceNotIn(List<String> values) {
            addCriterion("\"ACCOUNT_BALANCE\" not in", values, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_BALANCE\" between", value1, value2, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountBalanceNotBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_BALANCE\" not between", value1, value2, "accountBalance");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeIsNull() {
            addCriterion("\"ACCOUNT_FREEZE\" is null");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeIsNotNull() {
            addCriterion("\"ACCOUNT_FREEZE\" is not null");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeEqualTo(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" =", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeNotEqualTo(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" <>", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeGreaterThan(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" >", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeGreaterThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" >=", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeLessThan(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" <", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeLessThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" <=", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeLike(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" like", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeNotLike(String value) {
            addCriterion("\"ACCOUNT_FREEZE\" not like", value, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeIn(List<String> values) {
            addCriterion("\"ACCOUNT_FREEZE\" in", values, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeNotIn(List<String> values) {
            addCriterion("\"ACCOUNT_FREEZE\" not in", values, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_FREEZE\" between", value1, value2, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountFreezeNotBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_FREEZE\" not between", value1, value2, "accountFreeze");
            return (Criteria) this;
        }

        public Criteria andAccountDueIsNull() {
            addCriterion("\"ACCOUNT_DUE\" is null");
            return (Criteria) this;
        }

        public Criteria andAccountDueIsNotNull() {
            addCriterion("\"ACCOUNT_DUE\" is not null");
            return (Criteria) this;
        }

        public Criteria andAccountDueEqualTo(String value) {
            addCriterion("\"ACCOUNT_DUE\" =", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueNotEqualTo(String value) {
            addCriterion("\"ACCOUNT_DUE\" <>", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueGreaterThan(String value) {
            addCriterion("\"ACCOUNT_DUE\" >", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueGreaterThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_DUE\" >=", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueLessThan(String value) {
            addCriterion("\"ACCOUNT_DUE\" <", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueLessThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_DUE\" <=", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueLike(String value) {
            addCriterion("\"ACCOUNT_DUE\" like", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueNotLike(String value) {
            addCriterion("\"ACCOUNT_DUE\" not like", value, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueIn(List<String> values) {
            addCriterion("\"ACCOUNT_DUE\" in", values, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueNotIn(List<String> values) {
            addCriterion("\"ACCOUNT_DUE\" not in", values, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_DUE\" between", value1, value2, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountDueNotBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_DUE\" not between", value1, value2, "accountDue");
            return (Criteria) this;
        }

        public Criteria andAccountPaidIsNull() {
            addCriterion("\"ACCOUNT_PAID\" is null");
            return (Criteria) this;
        }

        public Criteria andAccountPaidIsNotNull() {
            addCriterion("\"ACCOUNT_PAID\" is not null");
            return (Criteria) this;
        }

        public Criteria andAccountPaidEqualTo(String value) {
            addCriterion("\"ACCOUNT_PAID\" =", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidNotEqualTo(String value) {
            addCriterion("\"ACCOUNT_PAID\" <>", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidGreaterThan(String value) {
            addCriterion("\"ACCOUNT_PAID\" >", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidGreaterThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_PAID\" >=", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidLessThan(String value) {
            addCriterion("\"ACCOUNT_PAID\" <", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidLessThanOrEqualTo(String value) {
            addCriterion("\"ACCOUNT_PAID\" <=", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidLike(String value) {
            addCriterion("\"ACCOUNT_PAID\" like", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidNotLike(String value) {
            addCriterion("\"ACCOUNT_PAID\" not like", value, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidIn(List<String> values) {
            addCriterion("\"ACCOUNT_PAID\" in", values, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidNotIn(List<String> values) {
            addCriterion("\"ACCOUNT_PAID\" not in", values, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_PAID\" between", value1, value2, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andAccountPaidNotBetween(String value1, String value2) {
            addCriterion("\"ACCOUNT_PAID\" not between", value1, value2, "accountPaid");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyIsNull() {
            addCriterion("\"TOTAL_MONEY\" is null");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyIsNotNull() {
            addCriterion("\"TOTAL_MONEY\" is not null");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyEqualTo(String value) {
            addCriterion("\"TOTAL_MONEY\" =", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyNotEqualTo(String value) {
            addCriterion("\"TOTAL_MONEY\" <>", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyGreaterThan(String value) {
            addCriterion("\"TOTAL_MONEY\" >", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyGreaterThanOrEqualTo(String value) {
            addCriterion("\"TOTAL_MONEY\" >=", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyLessThan(String value) {
            addCriterion("\"TOTAL_MONEY\" <", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyLessThanOrEqualTo(String value) {
            addCriterion("\"TOTAL_MONEY\" <=", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyLike(String value) {
            addCriterion("\"TOTAL_MONEY\" like", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyNotLike(String value) {
            addCriterion("\"TOTAL_MONEY\" not like", value, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyIn(List<String> values) {
            addCriterion("\"TOTAL_MONEY\" in", values, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyNotIn(List<String> values) {
            addCriterion("\"TOTAL_MONEY\" not in", values, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyBetween(String value1, String value2) {
            addCriterion("\"TOTAL_MONEY\" between", value1, value2, "totalMoney");
            return (Criteria) this;
        }

        public Criteria andTotalMoneyNotBetween(String value1, String value2) {
            addCriterion("\"TOTAL_MONEY\" not between", value1, value2, "totalMoney");
            return (Criteria) this;
        }
    }

    public static class Criteria extends GeneratedCriteria {

        protected Criteria() {
            super();
        }
    }

    public static class Criterion {
        private String condition;

        private Object value;

        private Object secondValue;

        private boolean noValue;

        private boolean singleValue;

        private boolean betweenValue;

        private boolean listValue;

        private String typeHandler;

        public String getCondition() {
            return condition;
        }

        public Object getValue() {
            return value;
        }

        public Object getSecondValue() {
            return secondValue;
        }

        public boolean isNoValue() {
            return noValue;
        }

        public boolean isSingleValue() {
            return singleValue;
        }

        public boolean isBetweenValue() {
            return betweenValue;
        }

        public boolean isListValue() {
            return listValue;
        }

        public String getTypeHandler() {
            return typeHandler;
        }

        protected Criterion(String condition) {
            super();
            this.condition = condition;
            this.typeHandler = null;
            this.noValue = true;
        }

        protected Criterion(String condition, Object value, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.typeHandler = typeHandler;
            if (value instanceof List<?>) {
                this.listValue = true;
            } else {
                this.singleValue = true;
            }
        }

        protected Criterion(String condition, Object value) {
            this(condition, value, null);
        }

        protected Criterion(String condition, Object value, Object secondValue, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.secondValue = secondValue;
            this.typeHandler = typeHandler;
            this.betweenValue = true;
        }

        protected Criterion(String condition, Object value, Object secondValue) {
            this(condition, value, secondValue, null);
        }
    }
}