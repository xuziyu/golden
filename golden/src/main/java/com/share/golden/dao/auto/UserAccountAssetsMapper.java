package com.share.golden.dao.auto;

import com.share.golden.entity.UserAccountAssets;
import com.share.golden.entity.UserAccountAssetsExample;
import java.math.BigDecimal;
import java.util.List;
import org.apache.ibatis.annotations.Param;

public interface UserAccountAssetsMapper {
    long countByExample(UserAccountAssetsExample example);

    int deleteByExample(UserAccountAssetsExample example);

    int deleteByPrimaryKey(BigDecimal accountId);

    int insert(UserAccountAssets record);

    int insertSelective(UserAccountAssets record);

    List<UserAccountAssets> selectByExample(UserAccountAssetsExample example);

    UserAccountAssets selectByPrimaryKey(BigDecimal accountId);

    int updateByExampleSelective(@Param("record") UserAccountAssets record, @Param("example") UserAccountAssetsExample example);

    int updateByExample(@Param("record") UserAccountAssets record, @Param("example") UserAccountAssetsExample example);

    int updateByPrimaryKeySelective(UserAccountAssets record);

    int updateByPrimaryKey(UserAccountAssets record);
}