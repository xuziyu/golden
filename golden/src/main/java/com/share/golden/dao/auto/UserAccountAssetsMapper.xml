<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
<mapper namespace="com.share.golden.dao.auto.UserAccountAssetsMapper">
  <resultMap id="BaseResultMap" type="com.share.golden.entity.UserAccountAssets">
    <id column="ACCOUNT_ID" jdbcType="DECIMAL" property="accountId" />
    <result column="USER_ID" jdbcType="VARCHAR" property="userId" />
    <result column="ACCOUNT_BALANCE" jdbcType="VARCHAR" property="accountBalance" />
    <result column="ACCOUNT_FREEZE" jdbcType="VARCHAR" property="accountFreeze" />
    <result column="ACCOUNT_DUE" jdbcType="VARCHAR" property="accountDue" />
    <result column="ACCOUNT_PAID" jdbcType="VARCHAR" property="accountPaid" />
    <result column="TOTAL_MONEY" jdbcType="VARCHAR" property="totalMoney" />
  </resultMap>
  <sql id="Example_Where_Clause">
    <where>
      <foreach collection="oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Update_By_Example_Where_Clause">
    <where>
      <foreach collection="example.oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Base_Column_List">
    "ACCOUNT_ID", "USER_ID", "ACCOUNT_BALANCE", "ACCOUNT_FREEZE", "ACCOUNT_DUE", "ACCOUNT_PAID", 
    "TOTAL_MONEY"
  </sql>
  <select id="selectByExample" parameterType="com.share.golden.entity.UserAccountAssetsExample" resultMap="BaseResultMap">
    select
    <if test="distinct">
      distinct
    </if>
    <include refid="Base_Column_List" />
    from USER_ACCOUNT_ASSETS
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
    <if test="orderByClause != null">
      order by ${orderByClause}
    </if>
  </select>
  <select id="selectByPrimaryKey" parameterType="java.math.BigDecimal" resultMap="BaseResultMap">
    select 
    <include refid="Base_Column_List" />
    from USER_ACCOUNT_ASSETS
    where "ACCOUNT_ID" = #{accountId,jdbcType=DECIMAL}
  </select>
  <delete id="deleteByPrimaryKey" parameterType="java.math.BigDecimal">
    delete from USER_ACCOUNT_ASSETS
    where "ACCOUNT_ID" = #{accountId,jdbcType=DECIMAL}
  </delete>
  <delete id="deleteByExample" parameterType="com.share.golden.entity.UserAccountAssetsExample">
    delete from USER_ACCOUNT_ASSETS
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </delete>
  <insert id="insert" parameterType="com.share.golden.entity.UserAccountAssets">
    insert into USER_ACCOUNT_ASSETS ("ACCOUNT_ID", "USER_ID", "ACCOUNT_BALANCE", 
      "ACCOUNT_FREEZE", "ACCOUNT_DUE", "ACCOUNT_PAID", 
      "TOTAL_MONEY")
    values (#{accountId,jdbcType=DECIMAL}, #{userId,jdbcType=VARCHAR}, #{accountBalance,jdbcType=VARCHAR}, 
      #{accountFreeze,jdbcType=VARCHAR}, #{accountDue,jdbcType=VARCHAR}, #{accountPaid,jdbcType=VARCHAR}, 
      #{totalMoney,jdbcType=VARCHAR})
  </insert>
  <insert id="insertSelective" parameterType="com.share.golden.entity.UserAccountAssets">
    insert into USER_ACCOUNT_ASSETS
    <trim prefix="(" suffix=")" suffixOverrides=",">
      <if test="accountId != null">
        "ACCOUNT_ID",
      </if>
      <if test="userId != null">
        "USER_ID",
      </if>
      <if test="accountBalance != null">
        "ACCOUNT_BALANCE",
      </if>
      <if test="accountFreeze != null">
        "ACCOUNT_FREEZE",
      </if>
      <if test="accountDue != null">
        "ACCOUNT_DUE",
      </if>
      <if test="accountPaid != null">
        "ACCOUNT_PAID",
      </if>
      <if test="totalMoney != null">
        "TOTAL_MONEY",
      </if>
    </trim>
    <trim prefix="values (" suffix=")" suffixOverrides=",">
      <if test="accountId != null">
        #{accountId,jdbcType=DECIMAL},
      </if>
      <if test="userId != null">
        #{userId,jdbcType=VARCHAR},
      </if>
      <if test="accountBalance != null">
        #{accountBalance,jdbcType=VARCHAR},
      </if>
      <if test="accountFreeze != null">
        #{accountFreeze,jdbcType=VARCHAR},
      </if>
      <if test="accountDue != null">
        #{accountDue,jdbcType=VARCHAR},
      </if>
      <if test="accountPaid != null">
        #{accountPaid,jdbcType=VARCHAR},
      </if>
      <if test="totalMoney != null">
        #{totalMoney,jdbcType=VARCHAR},
      </if>
    </trim>
  </insert>
  <select id="countByExample" parameterType="com.share.golden.entity.UserAccountAssetsExample" resultType="java.lang.Long">
    select count(*) from USER_ACCOUNT_ASSETS
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </select>
  <update id="updateByExampleSelective" parameterType="map">
    update USER_ACCOUNT_ASSETS
    <set>
      <if test="record.accountId != null">
        "ACCOUNT_ID" = #{record.accountId,jdbcType=DECIMAL},
      </if>
      <if test="record.userId != null">
        "USER_ID" = #{record.userId,jdbcType=VARCHAR},
      </if>
      <if test="record.accountBalance != null">
        "ACCOUNT_BALANCE" = #{record.accountBalance,jdbcType=VARCHAR},
      </if>
      <if test="record.accountFreeze != null">
        "ACCOUNT_FREEZE" = #{record.accountFreeze,jdbcType=VARCHAR},
      </if>
      <if test="record.accountDue != null">
        "ACCOUNT_DUE" = #{record.accountDue,jdbcType=VARCHAR},
      </if>
      <if test="record.accountPaid != null">
        "ACCOUNT_PAID" = #{record.accountPaid,jdbcType=VARCHAR},
      </if>
      <if test="record.totalMoney != null">
        "TOTAL_MONEY" = #{record.totalMoney,jdbcType=VARCHAR},
      </if>
    </set>
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByExample" parameterType="map">
    update USER_ACCOUNT_ASSETS
    set "ACCOUNT_ID" = #{record.accountId,jdbcType=DECIMAL},
      "USER_ID" = #{record.userId,jdbcType=VARCHAR},
      "ACCOUNT_BALANCE" = #{record.accountBalance,jdbcType=VARCHAR},
      "ACCOUNT_FREEZE" = #{record.accountFreeze,jdbcType=VARCHAR},
      "ACCOUNT_DUE" = #{record.accountDue,jdbcType=VARCHAR},
      "ACCOUNT_PAID" = #{record.accountPaid,jdbcType=VARCHAR},
      "TOTAL_MONEY" = #{record.totalMoney,jdbcType=VARCHAR}
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByPrimaryKeySelective" parameterType="com.share.golden.entity.UserAccountAssets">
    update USER_ACCOUNT_ASSETS
    <set>
      <if test="userId != null">
        "USER_ID" = #{userId,jdbcType=VARCHAR},
      </if>
      <if test="accountBalance != null">
        "ACCOUNT_BALANCE" = #{accountBalance,jdbcType=VARCHAR},
      </if>
      <if test="accountFreeze != null">
        "ACCOUNT_FREEZE" = #{accountFreeze,jdbcType=VARCHAR},
      </if>
      <if test="accountDue != null">
        "ACCOUNT_DUE" = #{accountDue,jdbcType=VARCHAR},
      </if>
      <if test="accountPaid != null">
        "ACCOUNT_PAID" = #{accountPaid,jdbcType=VARCHAR},
      </if>
      <if test="totalMoney != null">
        "TOTAL_MONEY" = #{totalMoney,jdbcType=VARCHAR},
      </if>
    </set>
    where "ACCOUNT_ID" = #{accountId,jdbcType=DECIMAL}
  </update>
  <update id="updateByPrimaryKey" parameterType="com.share.golden.entity.UserAccountAssets">
    update USER_ACCOUNT_ASSETS
    set "USER_ID" = #{userId,jdbcType=VARCHAR},
      "ACCOUNT_BALANCE" = #{accountBalance,jdbcType=VARCHAR},
      "ACCOUNT_FREEZE" = #{accountFreeze,jdbcType=VARCHAR},
      "ACCOUNT_DUE" = #{accountDue,jdbcType=VARCHAR},
      "ACCOUNT_PAID" = #{accountPaid,jdbcType=VARCHAR},
      "TOTAL_MONEY" = #{totalMoney,jdbcType=VARCHAR}
    where "ACCOUNT_ID" = #{accountId,jdbcType=DECIMAL}
  </update>
</mapper>